peer_review_data/main.py

```diff
@@ -1,41 +1,201 @@
 # -*- coding: utf-8 -*-
 import json
 from datetime import datetime, timedelta
+from typing import Dict
 
 from django.utils.timezone import utc
 
 from canvasData import *
+from peer_review_data import models
+from utils import dictSkipKeys
 
 LOGGER: Logger = getLogger(__name__)
 
 
+def saveCourseAndUsers(canvasCourse: CanvasCourse) -> (bool, Course):
+    course = models.Course.fromCanvasCourse(canvasCourse)
+    LOGGER.info(f'Saving {course}…')
+    course.save()
+
+    LOGGER.info(f'Saving users of {course}…')
+    for canvasUser in canvasCourse.get_users():
+        try:
+            user = models.User.fromCanvasUser(canvasUser)
+            LOGGER.debug(f'Saving {user}…')
+            user.save()
+        except Exception as e:
+            LOGGER.warning(f'Error saving {user}: {e}')
+
+    return True, course
+
+
+def saveSubmissions(canvasAssignment: CanvasAssignment):
+    canvasSubmissions: List[CanvasSubmission] = \
+        canvasAssignment.get_submissions()
+
+    for canvasSubmission in canvasSubmissions:
+        '''
+        The only untyped submission found during development was from
+        "Test Student".  It was discovered because that student wasn't
+        included with the rest of the course users and the relationship
+        between submission and missing user couldn't be created.  Testing
+        for untyped submissions eliminates those from "Test Student" and
+        possibly others.
+        '''
+        if canvasSubmission.submission_type is None:
+            LOGGER.warning('Skipping untyped submission '
+                           f'({canvasSubmission.id}) for '
+                           f'user ID ({canvasSubmission.user_id}).')
+            continue
+
+        try:
+            submission: models.Submission = \
+                models.Submission.fromCanvasSubmission(canvasSubmission)
+            LOGGER.debug(f'Saving {submission}…')
```

Comment:
instead of logging submission object only, can we also add {canvasSubmission.id}, {canvasSubmission.assignmet_id} and {canvasSubmission.user_id}?

Reply:
same comment applies to all save objects calls below, for logging more object details.

Reply:
Sure!  In fact, it already does that.  See [`Submission.__str()__` in `models.py`](https://github.com/lsloan/peer-review-data/blob/c8acfd061913ed23b2acd000522ba1162f4bf086/peer_review_data/models.py#L112-L114)

```python
    def __str__(self) -> str:
        return f'{self.__class__.__name__} ({self.id}): ' \
               f'({self.assignment}; {self.user})'
```

All the other classes do that, too.

Reply:
Good to know.